packages/loader/container-loader/src/loader.ts

```diff
@@ -51,53 +51,30 @@ function canUseCache(request: IRequest): boolean {
 }
 
 export class RelativeLoader extends EventEmitter implements ILoader {
-    // Because the loader is passed to the container during construction we need to resolve the target container
-    // after construction.
-    private readonly containerDeferred = new Deferred<Container>();
-
     /**
      * BaseRequest is the original request that triggered the load. This URL is used in case credentials need
      * to be fetched again.
      */
-    constructor(
-        private readonly loader: ILoader,
-        private readonly containerUrl: () => string | undefined,
-    ) {
+    constructor(private readonly loader: ILoader, private readonly container: Container) {
         super();
     }
 
     public get IFluidRouter(): IFluidRouter { return this; }
 
     public async resolve(request: IRequest): Promise<IContainer> {
         if (request.url.startsWith("/")) {
-            // If no headers are set that require a reload make use of the same object
-            const container = await this.containerDeferred.promise;
-            return container;
+            return this.container;
         }
 
         return this.loader.resolve(request);
     }
 
     public async request(request: IRequest): Promise<IResponse> {
-        const containerUrl = this.containerUrl();
         if (request.url.startsWith("/")) {
-            let container: IContainer;
-            if (canUseCache(request)) {
-                container = await this.containerDeferred.promise;
-            } else if (containerUrl === undefined) {
-                throw new Error("Container url is not provided");
-            } else {
-                container = await this.loader.resolve({ url: containerUrl, headers: request.headers });
-            }
-            return container.request(request);
+            return this.container.request(request);
```

Comment:
call this.resolve to get the container, and then request on that container

Reply:
But that does not check for cache. It just returns the cached container.